### DISCUSSION TITLE
this is not a planet？

### DISCUSSION THREAD

**Discussion by ggsnpdsn** [dec0dOS/zero-ui#162] [General]
I completely refer to this link https://github.com/dec0dOS/zero-ui#installation to install, but I checked through listpeers on the client and found that it is not a planet node, but a leaf node

root@ADF:~# zerotier-cli listpeers
200 listpeers <ztaddr> <path> <latency> <version> <role>
200 listpeers 62f865ae71 50.7.252.138/9993;6779;1383 392 - PLANET
**200 listpeers 7515da0fdd 120.241.29.164/14292;300;300 58 1.10.2 LEAF**
200 listpeers 778cde7190 103.195.103.66/9993;6780;1528 249 - PLANET
200 listpeers cafe04eba9 84.17.53.155/9993;6782;1495 277 - PLANET
200 listpeers cafe9efeb9 104.194.8.134/9993;1776;1568 209 - PLANET

Here only 7515da0fdd node is built by me, where do the rest of these extra planet nodes come from?

**Comment by sinamics** [comment]
Hello,

The output you've provided from zerotier-cli listpeers showcases a list of peers that your ZeroTier client is aware of. Here's a breakdown of what you're observing:

Your Node **7515da0fdd**: This is the node you set up, and as you noted, it's currently being recognized as a LEAF. A leaf node is a typical ZeroTier device, be it a server, computer, or any other device that's part of your ZeroTier network but isn't serving as a root or relay node.

Planet Nodes (e.g., 62f865ae71, 778cde7190, cafe04eba9, cafe9efeb9): ZeroTier uses a set of default global[ root servers known as "Planet"](https://zerotier.atlassian.net/wiki/spaces/SD/pages/7241732/Root+Server+IP+Addresses) nodes to assist in peer discovery and network coordination. When you install and start ZeroTier for the first time, your node will automatically discover and connect to these Planet nodes. They're part of the default ZeroTier configuration to ensure that new nodes can find and connect to existing networks.

So, in summary:

The node you built (7515da0fdd) being a LEAF is normal unless you specifically configured it to be a root server.
The other PLANET nodes you're seeing are default global roots provided by ZeroTier. They're expected to appear in your peer list and are an integral part of how ZeroTier operates.

**Comment by ggsnpdsn** [comment]
If I want the 7515da0fdd node to be a private root node, how do I set it up?
For example: I set up a server inside a large private network, and it cannot communicate with the internet. I completed the installation according to https://github.com/dec0dOS/zero-ui#installation and found that it does not work. It is a real leaf node Did not find him, where is the problem?

**Comment by sinamics** [comment]
If you want to create your own "world" without access to the internet, you need to create a root or planet server.
I would suggest looking at [mkworld](https://github.com/kmahyyg/ztncui-aio#usage); it's a tool written in Go for creating roots or planets.

Mkworld will generate a new planet.custom file based on your `mkworld.config.json` configuration. You'll need to distribute this file to your clients so they are aware of your new root server.
You also need to replace the file at /var/lib/zerotier-one/planet with your new planet.custom file, restart ZeroTier, and when you run `zerotier-cli listpeers`, all the ZeroTier roots will now be gone.

This should get you on the right track.
